src/main/java/com/mraof/minestuck/block/MSHorizontalDirectionalBlock.java

```diff
@@ -0,0 +1,48 @@
+package com.mraof.minestuck.block;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.*;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+
+import javax.annotation.Nullable;
+
+/**
+ * A copy of {@link HorizontalDirectionalBlock} that is no longer abstract
+ */
+public class MSHorizontalDirectionalBlock extends HorizontalDirectionalBlock {
```

Comment:
Yes

Reply:
Alright. Then it might be a good idea to look over the models for the knotted wood blocks. Right now only the top and bottom textures change with the directional rotation while the sides remain the same, unlike the glazed terracotta.

Reply:
yeah, noticed this but couldnt be bothered to try and fix it after the hassle it took to get it working as is

Reply:
may just make them a directional like pillars or logs and have the top use the side texture.